### ISSUE TITLE
decide how to treat intensity-weighted overlapping national/regional/local policies

### ISSUE THREAD

**Issue by bolliger32** [bolliger32/gpl-covid#67]
Do we sum? Do we take the max? See convo on slack. From @estherrolf:

[I’m gonna use US specific words so like state=adm1] specifically, 

(a) for non-popweighted variables, how do we aggregate state level contributions from county or city policies with differing policy_intensities? Like if a city has a policy with intensity 1 and a state level policy has intensity 0.2, is the state level (non-popweighted) intensity meant to be max(0.2,1) = 1 ?

(b) for popweighted values, are we counting state level contributions as min( sum_over_policies{percent_state_pop_effected * policy_intensity} , 1) ? If we do this, we would i think have to account for all the overlap in the percent_state_pop_effected and somewhoe also the intensities of the policies that overlap — seems prone to errors and likely that we’re not doing it the same way for each country.

basically I’m asking for us to come to a consensus on how to use policy_intensity  (I know you guys have all thought more about this than me, but as the person trying to implement the merge there’s a lot of complications arising). It’s possible theres a straightforward way to use it, it’s also possible that it’s not worth the trouble to account for fractional intensities (which maybe avoids encoding our judments on what fraction to give?).

**Comment by bolliger32** [comment]
We decided on the following (copied from Slack).

OK, so after getting a few responses to the above proposals and having folks clarify additional details that we need to iron out (thanks!), I propose the following approaches (in painful detail, just to make sure were on the same page :slightly_smiling_face: ). Apologies that some info is repeated from earlier, but hopefully with a bit more clarity:
We should be consistent in the way we apply intensity across admin units within a country and within a particular policy variable. That is to say, in order to calculate an intensity scale for a given policy variable, we should be looking at all the policies that get coded to this variable, regardless of the admin level at which they are implemented. Then, we should be determining how many "levels" there are and assigning intensities accordingly (e.g. if there are 3 levels of this policy, then we should be coding the intensity of those levels as .33, .67, 1 regardless of whether, for example, the middle intensity is only ever employed by provinces while the other two are national policies). This process should be done separately for the mandatory and optional versions of a variable. Since the optional stuff hasn't turned out to be super powerful and I'm not sure if we're including them in the regressions, a binary categorization (intensity=1 for all policies) seems fine if you don't want to go back through everything and align to this approach.
Policy intensity should be assigned based on the impact of that policy within a given day (i.e. "school is closed for a week" should be the same intensity as "school is closed for the rest of the year", b/c both entail "school is closed" on a given day)
When we have overlapping policies implemented at different admin levels:
a. For non "grab-bag" policy variables (e.g. not social_distance, and maybe a few others):
            i. For non-pop-weighted variables: Take the maximum of the mandatory policy intensities for all units lower (e.g. adm0) than, equal to, and higher (e.g. adm2) than the analysis unit (e.g. adm1) and assign it to the mandatory policy for the analysis unit. If no mandatory policies, take the maximum of the optional intensities and assign it to the optional policy variable for the analysis unit
            ii. For pop-weighted-variables: Take the maximum of the mandatory policy intensities for all units lower (e.g. adm0) than and equal to the analysis unit (e.g. adm1), and assign that as the "default" mandatory intensity for all units higher (e.g. adm2). If no mandatory policy at the analysis or lower unit, assign the maximum of the optional policies as the "default" optional intensity for all higher units. For any higher unit that has a specific policy, assign the appropriate version (mandatory or optional) of the policy variable at that higher unit the maximum of that intensity and the default intensity, with mandatory always taking priority over optional. For all that don't have a specific policy, assign them the default intensity (again, assigning this to the optional or mandatory version as appropriate). Then calculate the pop-weighting at the analysis unit level (e.g. adm1), separately for both optional and mandatory variables. Each higher unit should only have a non-0 intensity for optional or for mandatory (or neither), but not both.
        b. For "grab bag" policy variables like social distance: Each country can decide what makes the most sense to do here. If the specific policies employed at the various admin levels are actually the same policy, then the approach in (3a) is valid. If they are different policies within the same broad category, then we should add instead of taking the maximum, we should allow for both optional and mandatory policies, and we should otherwise follow the approach of (3a). This addition is appropriate across different admin levels b/c of (1). If some policies are the same and some are different, we will need a mixture of addition and taking the max. If we are adding, once the "processed" dataset has been calculated and formatted, the last step should be standardizing each variable so that it takes values between 0 and 1 (e.g. if the max from addition of sub-policies is 1.4, divide that entire column by 1.4). This standardization should again be done separately for each mandatory and optional version of each policy.
@dpa9694  has been working on a script that will implement this algorithm for USA and ITA, transforming [country]_policy_data_sources.csv into the policy variables in [country]_processed.csv. He will have a function arg called aggregate_vars which is a list of policy variables for which we should be adding (and allowing for both optional and mandatory simultaneously) instead of taking the max. If you'd like him to do the work for you, just make sure that your country's policy_data_sources.csv is in the standard format. Alternatively, you can write this transformation in your own script, since I think a lot of countries already have something that has a similar implementation to this one.
